pkg/scheduler/controller/shoot/scheduler_control.go

```diff
@@ -291,11 +327,23 @@ func getSeedWithLeastShootsDeployed(seedList []*gardencorev1beta1.Seed, shootLis
 	return bestCandidate, nil
 }
 
+func matchProvider(seed, shoot string, providers []string) bool {
```

Comment:
You could improve the naming

```suggestion
func matchProvider(seedProviderType, shootProviderType string, possibleProviders []string) bool {
```